lib/policyEvaluator/RequestContext.js

```diff
@@ -7,116 +7,28 @@ const parseIp = require('../ipCheck').parseIp;
-    completeMultipartUpload: 's3:PutObject',
-    initiateMultipartUpload: 's3:PutObject',
-    listMultipartUploads: 's3:ListBucketMultipartUploads',
-    listParts: 's3:ListMultipartUploadParts',
-    multipartDelete: 's3:AbortMultipartUpload',
-    objectDelete: 's3:DeleteObject',
-    objectDeleteVersion: 's3:DeleteObjectVersion',
-    objectDeleteTagging: 's3:DeleteObjectTagging',
-    objectDeleteTaggingVersion: 's3:DeleteObjectVersionTagging',
-    objectGet: 's3:GetObject',
-    objectGetVersion: 's3:GetObjectVersion',
-    objectGetACL: 's3:GetObjectAcl',
-    objectGetACLVersion: 's3:GetObjectVersionAcl',
-    objectGetTagging: 's3:GetObjectTagging',
-    objectGetTaggingVersion: 's3:GetObjectVersionTagging',
-    objectHead: 's3:GetObject',
-    objectPut: 's3:PutObject',
-    objectPutACL: 's3:PutObjectAcl',
-    objectPutACLVersion: 's3:PutObjectVersionAcl',
-    objectPutPart: 's3:PutObject',
-    objectPutTagging: 's3:PutObjectTagging',
-    objectPutTaggingVersion: 's3:PutObjectVersionTagging',
-    serviceGet: 's3:ListAllMyBuckets',
-    objectReplicate: 's3:ReplicateObject',
-};
+const {
+    actionMapRQ,
+    actionMapIAM,
+    actionMapSSO,
+    actionMapSTS,
+    actionMapMetadata,
+} = require('./utils/actionMaps');
 
 const _actionNeedQuotaCheck = {
     objectPut: true,
     objectPutPart: true,
 };
 
-const _actionMapIAM = {
-    attachGroupPolicy: 'iam:AttachGroupPolicy',
-    attachUserPolicy: 'iam:AttachUserPolicy',
-    createAccessKey: 'iam:CreateAccessKey',
-    createGroup: 'iam:CreateGroup',
-    createPolicy: 'iam:CreatePolicy',
-    createPolicyVersion: 'iam:CreatePolicyVersion',
-    createUser: 'iam:CreateUser',
-    deleteAccessKey: 'iam:DeleteAccessKey',
-    deleteGroup: 'iam:DeleteGroup',
-    deleteGroupPolicy: 'iam:DeleteGroupPolicy',
-    deletePolicy: 'iam:DeletePolicy',
-    deletePolicyVersion: 'iam:DeletePolicyVersion',
-    deleteUser: 'iam:DeleteUser',
-    detachGroupPolicy: 'iam:DetachGroupPolicy',
-    detachUserPolicy: 'iam:DetachUserPolicy',
-    getGroup: 'iam:GetGroup',
-    getGroupPolicy: 'iam:GetGroupPolicy',
-    getPolicy: 'iam:GetPolicy',
-    getPolicyVersion: 'iam:GetPolicyVersion',
-    getUser: 'iam:GetUser',
-    listAccessKeys: 'iam:ListAccessKeys',
-    listGroupPolicies: 'iam:ListGroupPolicies',
-    listGroups: 'iam:ListGroups',
-    listGroupsForUser: 'iam:ListGroupsForUser',
-    listPolicies: 'iam:ListPolicies',
-    listPolicyVersions: 'iam:ListPolicyVersions',
-    listUsers: 'iam:ListUsers',
-    putGroupPolicy: 'iam:PutGroupPolicy',
-    removeUserFromGroup: 'iam:RemoveUserFromGroup',
-};
-
-const _actionMapSSO = {
-    SsoAuthorize: 'sso:Authorize',
-};
-
-const _actionMapSTS = {
-    assumeRole: 'sts:AssumeRole',
-};
-
-const _actionMapMetadata = {
-    admin: 'metadata:admin',
-    default: 'metadata:bucketd',
-};
-
 function _findAction(service, method) {
     if (service === 's3') {
-        return _actionMap[method];
+        return actionMapRQ[method];
```

Comment:
isn't it `actionMapS3` that you want here?

Reply:
no, RQ is for the RequestContext file. S3 is used for all the methods in CloudServer